pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -118,27 +122,207 @@ func (pl *VolumeBinding) EventsToRegister() []framework.ClusterEventWithHint {
+				err = fmt.Errorf("waiting for ephemeral volume controller to create the persistentvolumeclaim %q", pvcName)
+				return false, err
+			}
+			return false, err
+		}
+
+		className := volume.GetPersistentVolumeClaimClass(pvc)
+		if className == "" {
+			return false, fmt.Errorf("no class for claim %q", pvcName)
+		}
+
+		if className == newSC.Name {
+			if !apiequality.Semantic.DeepEqual(newSC.AllowedTopologies, oldSC.AllowedTopologies) {
+				logger.V(4).Info("StorageClass was created or updated, and changed Provisioner", "AllowedTopologies", newSC.AllowedTopologies)
+				return true, nil
+			}
+		}
+
+		return false, nil
+	})
+
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	if result {
+		return framework.Queue, nil
+	}
+
+	logger.V(4).Info("StorageClass was created or updated, but it doesn't make this pod schedulable")
+	return framework.QueueSkip, nil
+}
+
+func (pl *VolumeBinding) isSchedulableAfterPersistentVolumeClaimChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	oldPVC, newPVC, err := util.As[*v1.PersistentVolumeClaim](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	logger = klog.LoggerWithValues(
+		logger,
+		"Pod", klog.KObj(pod),
+		"PersistentVolumeClaim", klog.KObj(newPVC),
+	)
+
+	result, err := processPodVolumesForQHint(pod, func(pvcName string, isEphemeral bool) (bool, error) {
+		if pvcName == newPVC.Name {
+			if newPVC.Status.Phase != oldPVC.Status.Phase {
+				logger.V(4).Info("PersistentVolumeClaim referenced by the Pod was created or updated, and changed Provisioner", "Phase", newPVC.Status.Phase)
+				return true, nil
+			}
+
+			if !apiequality.Semantic.DeepEqual(newPVC.Annotations, oldPVC.Annotations) {
+				logger.V(4).Info("PersistentVolumeClaim referenced by the Pod was created or updated, and changed Annotations")
+				return true, nil
+			}
+
+			if !apiequality.Semantic.DeepEqual(newPVC.Spec, oldPVC.Spec) {
+				logger.V(4).Info("PersistentVolumeClaim referenced by the Pod was created or updated, and changed Spec")
+				return true, nil
+			}
+		}
+
+		return false, nil
+	})
+
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	if result {
+		return framework.Queue, nil
+	}
+
+	logger.V(4).Info("PersistentVolumeClaim was created or updated, but it doesn't make this pod schedulable")
+	return framework.QueueSkip, nil
+}
+
+func (pl *VolumeBinding) isSchedulableAfterPersistentVolumeChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	_, newPV, err := util.As[*v1.PersistentVolume](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	logger = klog.LoggerWithValues(
+		logger,
+		"Pod", klog.KObj(pod),
+		"PersistentVolume", klog.KObj(newPV),
+	)
+
+	for _, vol := range pod.Spec.Volumes {
+		if vol.Name == newPV.Name {
+			logger.V(4).Info("PersistentVolume referenced by the Pod was created or updated")
+			return framework.Queue, nil
```

Comment:
If a PV was subject to processing by csi-translation-lib, it may undergo significant changes. Therefore, even if only a part of the PV changes, I believe it should be treated as a target for queueing. I would appreciate your thoughts on this.